src/app/messaging/component/message-window/message-window.component.ts

```diff
@@ -31,6 +34,9 @@ export class MessageWindowComponent implements OnInit, OnDestroy {
   constructor(private messagesService:MessagesService, 
     private authService: AuthenticateService,
     private route:ActivatedRoute) { 
+      this.route.paramMap.subscribe((paramMap) => {
```

Comment:
Why subscribe to the changes here?
Won't this component load when someone navigates to /1?
And won't this component die when someone navigates to /2 or some other route?
My point is this component will not live beyond 1 id here?
Maybe just look at snapshot

Reply:
done